classes/Carrier.php

```diff
@@ -741,7 +741,7 @@ public static function getCarriersForOrder($id_zone, $groups = null, $cart = nul
                 if ($row['range_behavior']) {
                     // Get id zone
                     if (!$id_zone) {
-                        $id_zone = (int) Country::getIdZone(Country::getDefaultCountryId());
+                        $id_zone = (int) Country::getIdZone($context->country->id);
```

Comment:
How is this even possible that this wasn't failing before 🤦 

It seems that this code wasn't working at all in this place.

Reply:
:+1:

Reply:
I don't think the country from the context is the default one However there is a `PS_COUNTRY_DEFAULT` configuration

So you could either use `Configruation::get('PS_COUNTRY_DEFAULT')` or create this `Country::getDefaultCountryId()` (which will itself get the configuration)